sentinelhub/sentinelhub_request.py

```diff
@@ -1,89 +1,201 @@
+        :type bounds: sentinelhub.BBox or sentinelhub.Geometry
+        """
+
+        if size is None and resolution is None:
+            raise ValueError("Either size or resolution argument should be given")
+
+        if not isinstance(evalscript, str):
+            raise ValueError("'evalscript' should be a string")
+
+        self.config = config or SHConfig()
+
+        self.mime_type = mime_type
+
+        self.payload = self.body(
+            request_bounds=self.bounds(bounds),
+            request_data=input_data,
+            request_output=self.output(size=size, resolution=resolution, responses=responses),
+            evalscript=evalscript
+        )
+
+        super().__init__(SentinelHubDownloadClient, **kwargs)
+
+    def get_data(self, save_data=False, redownload=False, max_threads=None):
+        """ Overrides DataRequest.get_data
+        """
+        self.download_list = [self._download_request]
+        return super().get_data(save_data=save_data, redownload=redownload, max_threads=max_threads)[0]
+
+    def create_request(self):
+        """ Prepares a download request
+        """
+        headers = {'content-type': 'application/json', "accept": self.mime_type.get_string()}
+
+        self._download_request = DownloadRequest(
+            request_type=RequestType.POST,
+            url=self.config.get_sh_processing_api_url(),
+            post_values=self.payload,
+            data_folder=self.data_folder,
+            hash_save=bool(self.data_folder),
+            data_type=self.mime_type,
+            headers=headers
+        )
+
+    @staticmethod
+    def input_data(data_source=None, time_interval=None, maxcc=1.0, mosaicking_order='mostRecent', other_args=None):
+        """ Generate request data
+        """
+
+        if not isinstance(data_source, DataSource):
+            raise ValueError("'data_source' should be an instance of sentinelhub.DataSource")
+
+        if not isinstance(maxcc, float) and (maxcc < 0 or maxcc > 1):
+            raise ValueError('maxcc should be a float on an interval [0, 1]')
+
+        if time_interval and not isinstance(time_interval, tuple):
+            raise ValueError("'time_interval should be a tupe of of (datetime, datetime) or (str, str)")
+
+        if time_interval and all(isinstance(time, str) for time in time_interval):
+            time_interval = datetime.fromisoformat(time_interval[0]), datetime.fromisoformat(time_interval[1])
+        elif time_interval and not all(isinstance(time, datetime) for time in time_interval):
+            raise ValueError("'time_interval should be a tupe of of (datetime, datetime) or (str, str)")
+
+        if time_interval:
+            date_from, date_to = time_interval
+            if date_from > date_to:
+                raise ValueError("'time_from' should not b greater than 'time_to'")
+            date_from, date_to = date_from.isoformat() + 'Z', date_to.isoformat() + 'Z'
+        else:
+            date_from, date_to = "", ""
```

Comment:
Instead of all this, you could probably use `parse_time_interval` function from `time_utils` module.

Reply:
Done.